### PR TITLE
remove redundant VMSS name substring suffix

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

This PR removes an unnecessary "-<resource type>" substring suffix that we include to name resources. We don't need to include the resource type string to identify the resource in the name, as the resource type is obviously implicit already.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: azure/services/scalesets/scalesets.go [hunk 1]
```diff
@@ -424,13 +424,13 @@ func (s *Service) buildVMSSFromSpec(ctx context.Context, vmssSpec azure.ScaleSet
 				NetworkProfile: &compute.VirtualMachineScaleSetNetworkProfile{
 					NetworkInterfaceConfigurations: &[]compute.VirtualMachineScaleSetNetworkConfiguration{
 						{
-							Name: to.StringPtr(vmssSpec.Name + "-netconfig"),
+							Name: to.StringPtr(vmssSpec.Name),
```
Comment (CecileRobertMichon) [inline azure/services/scalesets/scalesets.go:427]: is this a mutable property? what would happen for existing scale sets using this new code?

Comment (jackfrancis) [reply azure/services/scalesets/scalesets.go:427]: We should be very confident this isn't a mutable property.

Interestingly, this change should be fine due to expected `NetworkProfile` mutations via cloud-provider-azure during the lifecycle of the VMSS:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/scalesets/scalesets.go#L720

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Comment by CecileRobertMichon** [comment]
@jackfrancis can you please add a release note?

**Comment by jackfrancis** [comment]
@CecileRobertMichon done, thx!

**Pull request merged**